lib/std/os/linux.zig

```diff
@@ -1876,6 +1876,53 @@ pub fn fremovexattr(fd: usize, name: [*:0]const u8) usize {
     return syscall2(.fremovexattr, fd, @intFromPtr(name));
 }
 
+pub const sched_param = extern struct {
+    sched_priority: i32,
+};
+
+pub const SCHED = struct {
```

Comment:
this is an enum, not a struct?

Reply:
The problem with it being an enum is that the RESET_ON_FORK is not an independent value, but a flag that is ORed with one of the other values.

Reply:
I think the recommended thing to do would be to make it a `packed struct`, something like this:

```zig
pub const SCHED = packed struct(i32) {
  pub const Mode = enum(u3) {
    /// normal multi-user scheduling
    pub const OTHER = 0;
    /// FIFO realtime scheduling
    pub const FIFO = 1;
    /// Round-robin realtime scheduling
    pub const RR = 2;
    /// For "batch" style execution of processes
    pub const BATCH = 3;
    /// Low latency scheduling
    pub const ISO = 4;
    /// For running very low priority background jobs
    pub const IDLE = 5;
    /// Sporadic task model deadline scheduling
    pub const DEADLINE = 6;
  };
  mode: Mode, //bits [0, 2]
  _3: u27 = 0, //bits [3, 29]
  /// set to true to stop children from inheriting policies
  RESET_ON_FORK: bool = false, //bit 30
  _31: u1 = 0, //bit 31
}
```

Reply:
Thank you.  I will update my code accordingly.